### PR TITLE
Support handshake data validator

### PR DESCRIPTION
**Pull request by NicholasChan1025**
### What problem does this PR solve? 

Fix: #68 
解决issue68 自定义握手数据验证，验证失败则握手失败

### CODE REVIEW

**Review by lonng — COMMENTED** [review]
Please format your code.

File: cluster/handler.go [hunk 1]
```diff
@@ -252,6 +252,12 @@ func (h *LocalHandler) handle(conn net.Conn) {
 func (h *LocalHandler) processPacket(agent *agent, p *packet.Packet) error {
 	switch p.Type {
 	case packet.Handshake:
+
```
Comment (lonng) [inline cluster/handler.go:255]: Please remove the blank line.

File: internal/env/env.go [hunk 2]
```diff
@@ -49,12 +50,16 @@ var (
 	Serializer serialize.Serializer
 
 	GrpcOptions = []grpc.DialOption{grpc.WithInsecure()}
+	
+
```
Comment (lonng) [inline internal/env/env.go:54]: Why add two blank lines.

File: internal/env/env.go [hunk 3]
```diff
@@ -38,6 +38,7 @@ var (
 	CheckOrigin func(*http.Request) bool // check origin when websocket enabled
 	Debug       bool                     // enable Debug
 	WSPath      string                   // WebSocket path(eg: ws://127.0.0.1/WSPath)
+	HandshakeValidator func([]byte) error // When you need to verify the custom data of the handshake request
```
Comment (lonng) [inline internal/env/env.go:41]: Format code.

File: internal/env/env.go [hunk 4]
```diff
@@ -49,12 +50,16 @@ var (
...
 )
 
 func init() {
 	Die = make(chan bool)
 	Heartbeat = 30 * time.Second
 	Debug = false
 	CheckOrigin = func(_ *http.Request) bool { return true }
+	HandshakeValidator = func(_ []byte) error {return nil}
 	Serializer = protobuf.NewSerializer()
+
```
Comment (lonng) [inline internal/env/env.go:64]: Please remove the blank line.

**Comment by NicholasChan1025** [comment]
Format Code Success

**Comment by lonng** [comment]
@NicholasChan1025 Thanks for you contribution.

**Review by lonng — APPROVED** [review]
LGTM

**Pull request merged**